examples/src/multiplex/server.rs

```diff
@@ -23,7 +23,7 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let addr = "[::1]:50051".parse().unwrap();
 
     let greeter = GreeterServer::new(MyGreeter::default());
-    let echo = EchoServer::new(MyEcho::default());
+    let echo = EchoServer::new(MyEcho);
```

Comment:
It seems some supposed `default_constructed_unit_structs` targets are not detected, resulting in partial changing and a lack of consistency throughout the code. If it seems difficult to resolve, I think we can allow this rule for the examples as the crate level setting.

Reply:
What targets aren't detected? Personally I prefer being a little consistent about this over disabling a lint that we think points in the right direction.

Reply:
> What targets aren't detected?

The commented place here is one of them. `MyGreeter` is not detected.

I figured out the difference. The detected structs are defined like `struct MyEcho;` but not detected structs are done such as `struct MyGreeter {}` with empty brace style. I prefer using default trait to construct them, replacing the style with the empty brace `{}` one as the struct definitions, since it explicitly expresses constructing the value in the crates which do not effect on the efficiency.

Reply:
> What targets aren't detected?

The commented place here is one of them. `MyGreeter` is not detected.

I figured out the difference. The detected structs are defined like `struct MyEcho;` but not detected structs are done such as `struct MyGreeter {}` with empty brace style. I prefer using default trait to construct them in the places not affecting the efficiency, replacing the style with the empty brace `{}` one as the struct definitions, since it explicitly expresses constructing the value.

Reply:
Given that there is an actual difference, IMO further changes are off-topic for this PR.

Reply:
I do not agree just applying this clippy suggestion with introducing code inconsistency. When these changes are reverted, I agree merging this.

Reply:
It does not seem off-topic to me as the issues this change causes are relevant, and I do not agree just applying this clippy suggestion with introducing code inconsistency. When these changes are reverted, I agree merging this.